lib/generate.js

```diff
@@ -135,6 +135,18 @@ generate.javascript.builtinPostprocessors = {
     "id": "id"
 }
 
+generate.es = generate.ecmascript = function (parser, exportName) {
+    var output = "// Generated automatically by nearley, version " + parser.version + "\n";
+    output +=  "// http://github.com/Hardmath123/nearley\n";
+    output += "function id(x) { return x[0]; }\n";
+    output += parser.body.join('\n');
+    output += "let Lexer = " + parser.config.lexer + ";\n";
+    output += "let ParserRules = " + serializeRules(parser.rules, generate.javascript.builtinPostprocessors) + ";\n";
+    output += "let ParserStart = " + JSON.stringify(parser.start) + ";\n";
+    output += "export { Lexer, ParserRules, ParserStart };\n";
```

Comment:
I think we want to use `export default` here; importing these objects individually shouldn't be allowed.

Reply:
Are you sure about that? Named exports (the "individual objects") are generally considered best practice. Btw,

    export { Lexer, ParserRules, ParserStart }

is short for

    export { Lexer }
    export { ParserRules }
    export { ParserStart  }

and can be used just like you stated in [this comment](https://github.com/Hardmath123/nearley/pull/299#issuecomment-333079904):

    import * as generated from "./path"
    // generated.Lexer
    // generated.ParserRules
    // generated.ParserStart

Or alternatively:

    import { Lexer, ParserRules, ParserStart } from "./path"
    // Lexer
    // ParserRules
    // ParserStart

Reply:
Oh, and you probably want these to be consistent with TypeScript imports/exports (the syntax seems identical to me at first glance), which indeed use named exports and not default.

Reply:
The point is that we want to encourage use of `nearley.Grammar.fromCompiled`. If we were going for named exports, then I'd want it to be a single object (perhaps called "grammar"). But since the only object a generated grammar will _ever_ contain is the grammar, I think we should use a default export please. :-)

Reply:
Sure, I understand your reasoning, but it feels very wrong to use one approach for typescript and another for native imports/exports (since they both follow the same spec).

And changing ts to defaults would break existing code... Does that concern make sense?

Reply:
@tjvr Since this wasn't answered, I included both named and default exports.

Reply:
@tjvr Since you didn't respond to this, I included both default and named exports.

Reply:
Just the one default export please!

It’s true we don’t want to break backwards compatibility with TS, but we can always change TS later (perhaps in v3).